From: Peter Pentchev <roam@ringlet.net>
Date: Mon, 18 Mar 2024 12:13:12 +0200
Subject: Avoid implicit function definitions in the configure checks

Forwarded: not-yet
Last-Update: 2024-03-18

Add function prototypes and include system header files as needed.
---
 acinclude.m4  | 3 +++
 libscompat.m4 | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/acinclude.m4 b/acinclude.m4
index 4216e6f..bdc47ab 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -985,6 +985,9 @@ void binsearch(long startval, long endval);
 
 int resval = -1; /* -1: fail, 0: ok */
 
+void binsearch(long startval, long maxval);
+int checkval(long val);
+
 int
 main(void)
 {
diff --git a/libscompat.m4 b/libscompat.m4
index ba6975c..0c2de2c 100644
--- a/libscompat.m4
+++ b/libscompat.m4
@@ -34,6 +34,7 @@ AC_TRY_COMPILE([
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <stdio.h>
+#include <string.h>
 #include <sys/ioctl.h>
 #include <net/if.h>
 #include <string.h>
@@ -192,6 +193,7 @@ AC_MSG_CHECKING([for working sockatmark])
 AC_TRY_RUN([
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <stdio.h>
 
 int
 main()
@@ -419,6 +421,7 @@ AC_SEARCH_LIBS(clock_gettime, rt)
 AC_MSG_CHECKING([for CLOCK_MONOTONIC clock_gettime() support])
 AC_TRY_RUN([
 #include <sys/time.h>
+#include <stdio.h>
 #include <time.h>
 #include <stdio.h>
 
